app/tests/test_memory.py

```
@@ -0,0 +1,369 @@
+            
+            assert memory2.get_preference("test_pref") == "value1"
+            assert memory2.get_from_memory("test_key") == "test_value"
+
+    @patch("app.memory.context_manager.logger")
+    def test_error_handling_in_save(self, mock_logger, memory):
+        """Test error handling during save operations."""
+        # Make storage.save raise an exception
+        memory.storage.save = MagicMock(side_effect=Exception("Save failed"))
+        
+        # This should log error but not crash
+        with pytest.raises(Exception):
```

Ruff (0.11.9)

Do not assert blind exception: `Exception`

(B017)